### ISSUE TITLE
psiturk_shell.py's worker_bonus

### ISSUE THREAD

**Issue by jacob-lee** [NYUCCL/psiTurk#252]
I am wondering if not registering a bonus, when the bonus is 0, is the intended behavior:

```
if amount <= 0:
    print "bonus amount <=$0, no bonus given to", assignment_id
elif status == 7 and not override_status:
    print "bonus already awarded to ", assignment_id
else:
    success = self.amt_services.bonus_worker(assignment_id, amount, reason)
```

If a bonus of $0 is in the range of promised bonuses (e.g. bonus between $0 -$1), then a consequence of the code above is that assignment may appear to have been neglected by the requester; for example, a requester might look up all rows where status == 4, or 5 tto see who remains to be paid performance bonuses.

**Comment by deargle** [comment]
I think it's intentional. I can see arguments both ways. If a requester is manually looking up all rows where status == 4 or 5, wouldn't they also only query the rows where bonus > 0? Those are the only ones who would have not received their due.

**Comment by jacob-lee** [comment]
I can see your argument.

The Participant model's `bonus` column defaults to 0. I suspect this has
something to do with the decision in the bonus code above; alternatively,
it could be checked against null, if the bonus field were null prior to a
bonus value being explicitly assigned by the requester's code.

In my case, I was unsure whether a subject legitimately received a 0 for
their bonus, or whether a bonus was never assigned (given that the
Participant model's 'bonus' column defaults to 0); that the status was not
7 suggested to me that it was the latter... at least until I inspected the
code excerpted in my last message. Given the current state, and my task's
code, I did not have enough information to prove ex-post-fact which it was
that occurred in that instance.

Thanks!
